enhancements/cnf-tests-split.md

```diff
@@ -0,0 +1,204 @@
+2. Change the scope of the tests being run.
+
+## Proposal
+### User Stories
+#### Story 1
+
+As a developer, I want to run the cnf-tests suites so that I can validate my CNF feature works properly.
+
+#### Story 2
+As an Openshift administrator, I want to run the latency tests so that
+I can benchmark different CNF features.
+
+### Workflow Description
+
+This change will not affect the current API, including parameters/environment variables used to invoke the cnf-tests: `TEST_SUITES`, `FEATURES`, `FOCUS_TESTS`, `SKIP_TESTS`, `GINKGO_PARAMS`, `FAIL_FAST`, and `TESTS_REPORTS_PATH`. 
+However, with this proposal, instead of running test binaries that combine all the CNF components,  we will introduce an **optional** parameter named `CNF_COMPONENTS` to control what external test suites to run.  
+By default, the test will run all of the components it ran prior to this change.
+
+To explain this new variable further, read the following example:
+Running the `e2esuite` test suite, with an empty `CNF_COMPONENTS` and `FEATURES=metallb`, will result in 
+running ginkgo's test in all of the components: sriov, metallb, nto-performance, nto-latency, ptp and integration, adding the flag `-ginkgo.focus=metallb`.
+So in other words, the functionallity of the `FEATURES` var stays the same.
+
+When running the `e2esuite` test suite with `CNF_COMPONENTS=metallb` it will invoke only the metallb component
+of the e2esuite, i.e.: `cnf-tests/submodules/metallb-operator/test/e2e/functional/tests`
+and for the validation suite, it will run: `cnf-tests/submodules/metallb-operator/test/e2e/validation/tests`
+
+The mapping of suite + component to the external test suite path is done by:
+```bash
+declare -A TESTS_PATHS=\
+(["configsuite nto"]="cnf-tests/submodules/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/0_config"\
+ ["validationsuite cluster"]="cnf-tests/testsuites/validationsuite"
+ ["validationsuite metallb"]="cnf-tests/submodules/metallb-operator/test/e2e/validation"
+ ["cnftests integration"]="cnf-tests/testsuites/e2esuite"\
+ ["cnftests metallb"]="cnf-tests/submodules/metallb-operator/test/e2e/functional"\
+ ["cnftests sriov"]="cnf-tests/submodules/sriov-network-operator/test/conformance"\
+ ["cnftests nto-performance"]="cnf-tests/submodules/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance"\
+ ["cnftests nto-latency"]="cnf-tests/submodules/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/4_latency"\
+ ["cnftests ptp"]="cnf-tests/submodules/ptp-operator/test/conformance/serial")
+```
+
+To invoke the cnf-tests we'll use the same `Makefile` targets as before. e.g.: `functests-on-ci`, `feature-deploy-on-ci`, etc.
+Those rules will have a prerequisite to run the new `init-git-submodules` target which inits and updates the git submodules.
+The `init-git-submodules` will include the following env vars that control the commit hash of the components:
```

Comment:
Can we enforce that those submodules were inited and that their commit corresponds to what we have somewhere? It'd make sense to have a single source of truth for those hashes. This would prevent false negatives or us runnign the wrong set of tests.

Reply:
Yep, something like that

Reply:
Since we've decided on pointing to the latest commit, this conversation about validating we inited to the correct commit is outdated.